### PR TITLE
Prime1 - Fixes

### PR DESCRIPTION
**Pull request by toasterparty**
Note that this technically change the output from permalinks, but I'd argue it's still appropriate for a bugfix release as only room-rando permalinks which would have crashes due to incompatible room placements are affected.

I'd like to see this in a v4.2.2 release after a day or two of testing.

Also, this isn't all of the crashes, just the ones that were easy to fix. I should have a follow-up PR soon:TM:.

### CODE REVIEW

**Comment by henriquegemignani** [comment]
Hum, the db changes will cause older permalinks to just be incompatible so there'd be 4.2.1 permalinks and 4.2.2 permalinks. Just for prime I think, but still annoying.

Might be better to just call it 4.3.

File: randovania/games/prime1/layout/prime_configuration.py [hunk 1]
```diff
@@ -66,15 +66,24 @@ def dangerous_settings(self) -> List[str]:
...
         if self.room_rando != RoomRandoMode.NONE:
-            result.append("Room randomizer")
-        
+            result.append("Room Randomizer")
+
         if self.large_samus:
             result.append("Large Samus")
 
+        if self.superheated_probability > 0:
+            result.append("Extra Superheated Rooms")
```
Comment (henriquegemignani) [inline randovania/games/prime1/layout/prime_configuration.py:81]: Should add a test that calls `dangerous_settings` on a preset with all these options enabled.

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

File: randovania/games/prime1/exporter/patch_data_factory.py [hunk 2]
```diff
@@ -555,11 +555,13 @@ def pick_random_dst(src_name, src_dock):
 
         boss_sizes = None
         if self.configuration.random_boss_sizes:
-            def get_random_size(min, max):
+            def get_random_size(minimum, maximum):
                 if self.rng.choice([True, False]):
-                    return self.rng.uniform(min, 0.8)
+                    temp = [self.rng.uniform(minimum, 1.0), self.rng.uniform(minimum, 1.0)]
```
Comment (henriquegemignani) [inline randovania/games/prime1/exporter/patch_data_factory.py:560]: taking 2RN so it bias towards smallers numbers?

Comment (toasterparty) [reply randovania/games/prime1/exporter/patch_data_factory.py:560]: Correct. In testing "real random" and "`< 0.8` / `> 1.2` random" felt less exciting.

File: randovania/games/prime1/layout/prime_configuration.py (hunk 1 above)
Comment (toasterparty) [reply randovania/games/prime1/layout/prime_configuration.py:81]: To test that a preset with all dangerous options is flagged as dangerous? I feel like that's not in jeopardy of failing ever

File: CHANGELOG.md [hunk 3]
```diff
@@ -23,11 +23,21 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Metroid Prime
 
-- Nothing.
+- Fixed: Random Superheated, Random Submerged and Dangerous Gravity Suit logic now trigger dialog warning in Multiword sessions
+- Fixed: Reduced Ridley's max size
+- Fixed: Reduced Thardus' max size
```
Comment (henriquegemignani) [inline CHANGELOG.md:28]: Shouldn't there also be a mention for incinerator drone?

Comment (henriquegemignani) [reply CHANGELOG.md:28]: And another for omega pirate minimum size

File: test/games/prime1/exporter/test_prime_dangerous_settings.py [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import pytest
+
+from randovania.interface_common.players_configuration import PlayersConfiguration
+from randovania.layout.layout_description import LayoutDescription
+
+def test_dangerous_settings(test_files_dir, rdvgame_filename="prime1_crazy_seed.rdvgame"):
+    rdvgame = test_files_dir.joinpath("log_files", rdvgame_filename)
+    layout_description = LayoutDescription.from_file(rdvgame)    
+    players_config = PlayersConfiguration(0, {0: "Prime", 1: "Echoes"})
```
Comment (henriquegemignani) [inline test/games/prime1/exporter/test_prime_dangerous_settings.py:9]: This doesn't seem necessary

File: test/games/prime1/exporter/test_prime_dangerous_settings.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+    preset = layout_description.get_preset(players_config.player_index)
+    
+    print(preset.dangerous_settings())
```
Comment (henriquegemignani) [inline test/games/prime1/exporter/test_prime_dangerous_settings.py:12]: Extra print

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

File: randovania/games/prime1/layout/prime_configuration.py (hunk 1 above)
Comment (toasterparty) [reply randovania/games/prime1/layout/prime_configuration.py:81]: The test I added was in response to this comment.

File: test/games/prime1/exporter/test_prime_dangerous_settings.py (hunk 4 above)
Comment (henriquegemignani) [reply test/games/prime1/exporter/test_prime_dangerous_settings.py:9]: Just `players_config`

**Review by henriquegemignani — APPROVED** [review]

**Pull request merged**